commit 7114c6f237bcf35fad7f4531deb3dbd6507d9a83 (refs/changes/18/4018/3)
Author: Jason King <jason.king@joyent.com>
Date:   2018-05-30T12:14:35-05:00 (1 year, 4 months ago)
    
    OS-6980 libvarpd leaks varpd_query_ts
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Reviewed by: Dan McDonald <danmcd@joyent.com>
    Approved by: Dan McDonald <danmcd@joyent.com>

diff --git a/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c b/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
index b7506c56cf..124e3c5791 100644
--- a/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
+++ b/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
@@ -514,6 +514,8 @@ libvarpd_plugin_query_reply(varpd_query_handle_t *vqh, int action)
 	else
 		libvarpd_panic("plugin %s passed in an invalid action: %d",
 		    vqp->vq_instance->vri_plugin->vpp_name, action);
+
+	umem_cache_free(vqp->vq_instance->vri_impl->vdi_qcache, vqp);
 }
 
 void
